extension/chrome/elements/compose-modules/formatters/encrypted-mail-msg-formatter.ts

```diff
@@ -22,19 +22,49 @@ import { AcctStore } from '../../../../js/common/platform/store/acct-store.js';
 import { FcUuidAuth } from '../../../../js/common/api/account-servers/flowcrypt-com-api.js';
 import { SmimeKey } from '../../../../js/common/core/crypto/smime/smime-key.js';
 import { PgpHash } from '../../../../js/common/core/crypto/pgp/pgp-hash.js';
+import { UploadedMessageData } from '../../../../js/common/api/account-server.js';
+import { ParsedKeyInfo } from '../../../../js/common/core/crypto/key-store-util.js';
+import { MultipleMessages } from './general-mail-formatter.js';
 
 export class EncryptedMsgMailFormatter extends BaseMailFormatter {
 
-  public sendableMsg = async (newMsg: NewMsgData, pubkeys: PubkeyResult[], signingPrv?: Key): Promise<SendableMsg> => {
+  public sendableMsgs = async (newMsg: NewMsgData, pubkeys: PubkeyResult[], signingKey?: ParsedKeyInfo): Promise<MultipleMessages> => {
     if (newMsg.pwd && !this.isDraft) {
       // password-protected message, temporarily uploaded (already encrypted) to:
       //    - flowcrypt.com/api (consumers and customers without on-prem setup), or
       //    - FlowCrypt Enterprise Server (enterprise customers with on-prem setup)
       //    It will be served to recipient through web
-      const { url: msgUrl, externalId } = await this.prepareAndUploadPwdEncryptedMsg(newMsg); // encrypted for pwd only, pubkeys ignored
+      const uploadedMessageData = await this.prepareAndUploadPwdEncryptedMsg(newMsg); // encrypted for pwd only, pubkeys ignored
```

Comment:
@tomholub Do I understand correctly that we need to send all recipients to FES (including those who have pubkeys and won't be sent individual messages to)? FES meanwhile generates and returns  personalized `externalId`s for every recipient, some of them will not be used.

Reply:
That's a design mistake, I think. @jerbell any thoughts?

For now, yes - please send all recipients, but only use some of the results that actually need it.

Reply:
It feels like it has been a while since I've looked at this.  I'll try to give the correct answer.

I assume we are talking about the initial message send/post.

It is true that an external id is created for each To/Cc/Bcc recipient.  The overhead isn't that great, and the difference is an empty or filled field in the DB.  I can see how it might be misleading though.  If you didn't want this we would need to change the `MessageUploadRequest` object - perhaps I considered this the least disruptive course at the time.

We need to send all recipients though in case there is a reply all through the web portal.

If you see this as a design flaw then we could add the field in a list of `passwordEmails` so I would know who to generate & save external ids for.  The response would then filter out the `null` external ids (this is happening already).

Reply:
Thanks - I've filed issue to improve that later. @rrrooommmaaa for now, let's use the API as is, and do your own filtering on the result as you said.